peer/brontide.go

```diff
@@ -104,13 +112,13 @@ type TimestampedError struct {
 	Timestamp time.Time
 }
 
-// peer is an active peer on the Lightning Network. This struct is responsible
+// Brontide is an active peer on the Lightning Network. This struct is responsible
 // for managing any channel state related to this peer. To do so, it has
 // several helper goroutines to handle events such as HTLC timeouts, new
 // funding workflow, and detecting an uncooperative closure of any active
 // channels.
 // TODO(roasbeef): proper reconnection logic
-type peer struct {
+type Brontide struct {
```

Comment:
I think this name works mainly to avoid having `peer.Peer` everywhere, but once this is merged we could go with another name since this isn't _strictly_ tied to brontide, though it does assert that specific connection from a regular `net.Conn`.

Reply:
Yeah I wasn't sure about the name, also there's an `lnpeer.Peer` interface that it implements. I thought it would get too wordy.

Reply:
Yeah I think it’s fine to drop `net.Conn` and focus only on `*brontide.Conn`, you can’t really use anything else because the framing semantics are different from regular connections

Reply:
Does having a `net.Conn` make testing any easier? That's the *only* reason I considered keeping it around. But if all MUST be *brontide.Conn what's the point

Reply:
no not really, a `brontide.Conn` implements `net.Conn` in case you wanted to use it in a separate context. in the peer however we don't use `Read` or `Write` because the protocol operates at the _message_ to handle timeouts properly. so in reality we only use `ReadMessage`, `WriteMessage`, and `Flush`